Sprint_6

BarkhatovNikolay opened the pull request:
Я запутался как правильно отправить отчет. Вроде бы правильно все поправил

Commit:
задание по спринту 6 + отчет jacoco

okryukova commented on src/test/java/com/example/TestLion.java:
```diff
@@ -0,0 +1,59 @@
+    public void testGetKittens() throws Exception {
+        when(feline.getKittens()).thenReturn(1);
+        Lion lion = new Lion("Самец", feline);
+        assertEquals("Должен возвращать кол-во котят = ", 1, lion.getKittens());
```
⛔️Нужно исправить. Для юнит-тестов применим подход: один тест, значит одна проверка. В этом тесте две проверки (Mockito.verify, assertEquals), а должна быть одна. Исправь, пожалуйста, этот момент во всем коде.

okryukova commented on src/test/java/com/example/TestLion.java:
```diff
@@ -0,0 +1,59 @@
+        verify(feline, times(1)).getKittens();
+    }
+    @Test
+    public void testLionHasManeMale() throws Exception {
```
⛔️Нужно исправить.  Тесты на пол льва есть в параметризированном классе. Не нужно их дублировать

okryukova commented on src/test/java/com/example/TestAnimal.java:
```diff
@@ -0,0 +1,33 @@
+public class TestAnimal {
+    @Test
+    public void getFoodForHerbivoresTest() throws Exception {
+        Animal animal = new Animal();
```
⚠️Можно улучшить. Инициализацию объекта можно вынести в метод с аннотацией before

okryukova commented on src/test/java/com/example/TestFelineWithParameters.java:
```diff
@@ -0,0 +1,34 @@
+        this.expectedKittenCounts = expectedKittenCounts;
+    }
+
+    @Parameterized.Parameters
```
⚠️Можно улучшить. В параметризованных тестах для аннотации Parameterized.Parameters лучше использовать аргумент name: @Parameterized.Parameters(name = "Тестовые данные: {0} {1}"), где {0}, {1} - индексы параметров. Это повысит информативность проверки

Commit:
1)Инициализацимя объекта вынесена в метод с анотацией Before
2)Использован аргумент name: @Parameterized.Parameters(name = "Тестовые данные: {0} {1}")
3)Разделены на разные тесты две проверки (Mockito.verify, assertEquals)
4)Дублирование тестов на пол льва удалены из TestLion

okryukova commented on src/test/java/com/example/TestCat.java:
```diff
@@ -0,0 +1,47 @@
+        List<String> expectedFood = Arrays.asList("Животные", "Птицы", "Рыба");
+        when(feline.eatMeat()).thenReturn(expectedFood);
+        List<String> actualFood = cat.getFood();
+        Assert.assertEquals("getFood должен возвращать еду для хищников от Feline", expectedFood, actualFood);
```
⛔️Нужно исправить. Для юнит-тестов применим подход: один тест, значит одна проверка. В этом тесте две проверки (Mockito.verify, assertEquals), а должна быть одна. Исправь, пожалуйста, этот момент во всем коде.
